aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -0,0 +1,151 @@
+          },
+          [](Vec base) -> Vec { return base.sqrt(); }
+        );
+      } else if (exp == 2) {
+        cpu_kernel_vec(iter,
+          [](scalar_t base) -> scalar_t { return base * base; },
+          [](Vec base) -> Vec { return base * base; }
+        );
+      } else if (exp == 3) {
+        cpu_kernel_vec(iter,
+          [](scalar_t base) -> scalar_t { return base * base * base; },
+          [](Vec base) -> Vec { return base * base * base; }
+        );
+      } else if (exp == -0.5) {
+        cpu_kernel_vec(iter,
+          [](scalar_t base) -> scalar_t {
+            return 1.0 / std::sqrt((long double)base);
+          },
+          [](Vec base) -> Vec { return base.sqrt().reciprocal(); }
+        );
+      } else if (exp == -1) {
+        cpu_kernel_vec(iter,
+          [](scalar_t base) -> scalar_t { return 1.0 / base; },
+          [](Vec base) -> Vec { return base.reciprocal(); }
+        );
+      } else if (exp == -2) {
+        cpu_kernel_vec(iter,
+          [](scalar_t base) -> scalar_t { return 1.0 / (base * base); },
+          [](Vec base) -> Vec { return (base * base).reciprocal(); }
+        );
+      } else {
+        cpu_kernel_vec(iter,
+          [=](scalar_t base) -> scalar_t {
+            return std::pow((long double)base, exp);
+          },
+          [=](Vec base) -> Vec { return base.pow(exp); }
+        );
+      }
+    });
+  } else {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "pow", [&]() {
+      auto exp = exp_scalar.to<double>();
+      if (exp == 0.5) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return std::sqrt(base); }
+        );
+      } else if (exp == 2) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return base * base; }
+        );
+      } else if (exp == 3) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return base * base * base; }
+        );
+      } else if (exp == -0.5) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return 1.0 / std::sqrt(base); }
+        );
+      } else if (exp == -1) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return 1.0 / base; }
+        );
+      } else if (exp == -2) {
+        cpu_kernel(iter,
+          [](scalar_t base) -> scalar_t { return 1.0 / (base * base); }
+        );
+      } else {
+        cpu_kernel(iter,
+          [=](scalar_t base) -> scalar_t {
+            return std::pow((long double)base, exp);
+          }
+        );
+      }
+    });
+  }
+}
+
+void pow_scalar_tensor_kernel(TensorIterator& iter, Scalar base_scalar) {
+  if (base_scalar.isFloatingPoint()) {
+    const auto base = base_scalar.to<double>();
+    AT_DISPATCH_ALL_TYPES(iter.dtype(), "pow", [&]() {
+      cpu_kernel(iter,
+        [=](scalar_t exp) -> scalar_t {
+          return std::pow((long double)base, exp);
+        }
+      );
+    });
+  } else {
+    const auto base = base_scalar.to<int64_t>();
```

Comment:
There is no difference in dispatch for floating and integral. Just define final base value as `long double` and pass it to common dispatch.